tests/apollo/test_skvbc_s3_integrity_check.py

```diff
@@ -0,0 +1,202 @@
+        """
+        Start integrity check
+        """
+        with log.start_action(action_type="start_integrity_check"):
+            stdout_file = None
+            stderr_file = None
+
+            if os.environ.get('KEEP_APOLLO_LOGS', "").lower() in ["true", "on"]:
+                test_name = os.environ.get('TEST_NAME')
+
+                if not test_name:
+                    now = datetime.now().strftime("%y-%m-%d_%H:%M:%S")
+                    test_name = f"{now}_{bft_network.current_test}"
+
+                test_dir = f"{bft_network.builddir}/tests/apollo/logs/{test_name}/{bft_network.current_test}/"
+                test_log = f"{test_dir}stdout_integrity_check.log"
+                log.log_message(message_type=f"test log is: {test_log}")
+                os.makedirs(test_dir, exist_ok=True)
+
+                stdout_file = open(test_log, 'w+')
+                stderr_file = open(test_log, 'w+')
+
+                stdout_file.write("############################################\n")
+                stdout_file.flush()
+                stderr_file.write("############################################\n")
+                stderr_file.flush()
+
+                s3_config_path = os.path.join(bft_network.builddir, s3_config_file)
+                integrity_check_fds = (stdout_file, stderr_file)
+                integrity_check_exe = os.path.join(bft_network.builddir, "storage", "s3_integrity_check")
+                integrity_check_cmd = [integrity_check_exe,
+                                       "-k", keys_file,
+                                       "-3", s3_config_path]
+                if key_to_validate is not None:
+                    integrity_check_cmd.append("-v")
+                    integrity_check_cmd.append(key_to_validate)
+                else:
+                    integrity_check_cmd.append("-a")
+                log.log_message(message_type="starting the subprocess")
+                integrity_check_pid = subprocess.Popen(
+                    integrity_check_cmd,
+                    stdout=stdout_file,
+                    stderr=stderr_file,
+                    close_fds=True)
+                try:
+                    exit_code = integrity_check_pid.wait()
+                    assert exit_code == 0
+                except Exception as e:
+                    assert False
+                finally:
+                    for fd in integrity_check_fds:
+                        fd.close()
+
+    def _delete_block(self):
+        client = Minio("127.0.0.1:9000", access_key="concordbft", secret_key="concordbft", secure=False)
+        buckets = client.list_buckets()
+        bucketName = buckets[0].name
+        try:
+            client.remove_object(bucketName, 'concord/blocks/1')
+        except ResponseError as err:
+            assert False
+
+    async def _wait_for_st(self, bft_network, ro_replica_id, seqnum_threshold=150):
+        # TODO replace the below function with the library function:
```

Comment:
why not TODO now? :-)

Reply:
removed this function and used `await bft_network.wait_for_state_transfer_to_start()` instead